contracts/v6/OraclePrice.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Unlicensed
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./interface/KeeperCompatibleInterface.sol";
+import "./interface/IOmsPolicy.sol";
+import "./interface/EACAggregatorProxy.sol";
+import "./library/SafeMath.sol";
+import "./common/Ownable.sol";
+
+contract OraclePrice is Ownable, KeeperCompatibleInterface {
+    using SafeMath for uint256;
+
+    struct OracleInfo {
+        address oracleAddress;
+        bool status;
+        bytes32 symbolHash;
+        int256 lastPrice; 
+    }
+    
+    OracleInfo[] public oracleInfo;
+    address public policyContract;
+    uint256 public deviationThreshold;
+    uint public immutable interval;
+    uint public lastTimeStamp;
+    uint public counter;
+    
+    constructor(OracleInfo[] memory _oracles, address _policyContract, uint256 _deviationThreshold, uint _updateInterval) public {
+        policyContract = _policyContract;
+        deviationThreshold = _deviationThreshold;
+        
+        for(uint256 i=0; i<_oracles.length; i++) {
```

Comment:
Perhaps I'm missing something with how solidity works but why recreate loop over the OracleInfo objects and not just store _oracles straight into the oracleInfo field?

Would that cause issues somehow?

Reply:
Yes we can add oracles on deployment time and if you want then i can remove this from constructor and we can add oracle's after deployment.

Reply:
The question is more about replacing

`for(uint256 i=0; i<_oracles.length; i++) {
            OraclePriceStruct.OracleInfo memory oracles = _oracles[i];
            oracleInfo.push(OraclePriceStruct.OracleInfo({
                oracleAddress: oracles.oracleAddress,
                status: oracles.status,
                symbolHash: oracles.symbolHash,
                lastPrice: 0
            }));
        }`

with 

`oracleInfo = _oracles`;

Reply:
Did you end up figuring out if you can simplify this bit?

Reply:
Just about to test that now.

Reply:
Doesn't work. If I use oracleInfo = _oracles, solidity complains about a memory to storage error. 
Will leave for now and think about simplifying this further when we upgrade the solidity version.